packages/build/src/tarball.spec.ts

```diff
@@ -0,0 +1,133 @@
+import os from 'os';
+import fs from 'fs';
+import path from 'path';
+import { expect } from 'chai';
+import BuildVariant from './build-variant';
+import {
+  tarball,
+  tarballPath,
+  tarballPosix,
+  tarballDebian,
+  tarballWindows
+} from './tarball';
+
+describe('tarball module', () => {
+  describe('.tarballPath', () => {
+    context('when the build variant is windows', () => {
+      it('returns the windows tarball name', () => {
+        expect(tarballPath('', BuildVariant.Windows, '1.0.0')).
+          to.equal('mongosh-1.0.0-win32.zip');
+      });
+    });
+
+    context('when the build variant is macos', () => {
+      it('returns the tarball name', () => {
+        expect(tarballPath('', BuildVariant.MacOs, '1.0.0')).
+          to.equal('mongosh-1.0.0-darwin.zip');
+      });
+    });
+
+    context('when the build variant is linux', () => {
+      it('returns the tarball name', () => {
+        expect(tarballPath('', BuildVariant.Linux, '1.0.0')).
+          to.equal('mongosh-1.0.0-linux.tgz');
+      });
+    });
+
+    context('when the build variant is debian', () => {
+      it('returns the tarball name', () => {
+        expect(tarballPath('', BuildVariant.Debian, '1.0.0')).
+          to.equal('mongosh_1.0.0_amd64.deb');
+      });
+    });
+  });
+
+  describe('.tarballPosix', () => {
+    const version = '1.0.0';
+    const expectedTarball = tarballPath(__dirname, BuildVariant.Linux, version);
+
+    before(() => {
+      return tarballPosix(__dirname, expectedTarball);
+    });
+
+    after((done) => {
+      fs.unlink(expectedTarball, done);
+    });
+
+    it('builds the executable', (done) => {
+      fs.stat(expectedTarball, (error) => {
+        expect(error).to.equal(null);
+        done();
+      });
+    });
+  });
+
+  // macos and windows build variants does not come with installed 'dpkg' so
+  // this test fails. Run this test only on linux.
+  if (os.platform() === BuildVariant.Linux) {
```

Comment:
Why do we compare a platform with a BuildVariant?

Shouldn't this be `process.env.BUILD_VARIANT === BuildVariant.Linux`

Reply:
for *some* reason `process.env.BUILD_VARIANT` is not available here. I agree, this is a bit of hacky statement, and it's on actually dependent on the build variant, but the current platform (i.e. it can run on any linux). I can do either of these two instead:
```js
if (os.platform() === 'linux') {
```
_or_:
```
import Platform from './platform';
if (os.platform() === Platform.Linux
```